crates/fuel-core/src/p2p_test_helpers.rs

```diff
@@ -587,9 +590,38 @@ impl Node {
                     result = wait_transaction.next() => {
                         let status = result.unwrap().unwrap();
 
-                        if matches!(status, TransactionStatus::Failure { .. })
-                            || matches!(status, TransactionStatus::Success { .. }) {
-                            break
+                        match (latest_height, status.inclusion_height()) {
+                            (Some(ref mut latest_height), Some(inclusion_height)) => {
+                                if inclusion_height > *latest_height {
+                                    *latest_height = inclusion_height;
+                                }
+                                break;
+                            }
+                            (None, Some(inclusion_height)) => {
+                                latest_height = Some(inclusion_height);
+                                break;
+                            }
+                            _ => {}
+                        }
+                    }
+                    _ = self.node.await_shutdown() => {
+                        panic!("Got a stop signal")
+                    }
+                }
+            }
+        }
+
+        if let Some(latest_height) = latest_height {
```

Comment:
I think it will be better if GasPriceService provided a way to wait until block height `X`, like `Relayer` or `OffChainWorker` do.

We have several tests that could be more stable if we had this functionality.